plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteTypeMapping.java

```diff
@@ -0,0 +1,387 @@
+                .addRoundTrip("date", "DATE '2017-07-01'", DATE, "DATE '2017-07-01'") // summer on northern hemisphere (possible DST)
+                .addRoundTrip("date", "DATE '2017-01-01'", DATE, "DATE '2017-01-01'") // winter on northern hemisphere (possible DST on southern hemisphere)
+                .addRoundTrip("date", "DATE '1983-04-01'", DATE, "DATE '1983-04-01'") // change forward at midnight in Vilnius
+                .addRoundTrip("date", "DATE '1983-10-01'", DATE, "DATE '1983-10-01'")
+                .addRoundTrip("date", "DATE '2983-10-01'", DATE, "DATE '2983-10-01'")
+                .addRoundTrip("date", "DATE '3983-10-01'", DATE, "DATE '3983-10-01'")
+                .addRoundTrip("date", "DATE '4983-10-01'", DATE, "DATE '4983-10-01'")
+                .addRoundTrip("date", "DATE '9983-10-01'", DATE, "DATE '9983-10-01'")
+                .addRoundTrip("date", "DATE '9999-01-01'", DATE, "DATE '9999-01-01'")
+                .addRoundTrip("date", "DATE '9999-12-31'", DATE, "DATE '9999-12-31'")
+                .execute(getQueryRunner(), session, trinoCreateAndInsert("test_date"))
+                .execute(getQueryRunner(), session, igniteCreateAndInsert("test_date"));
+    }
+
+    // Ignite char is mapped to varchar
+    @Test
+    public void testChar()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("char(10)", "'NULL'", createVarcharType(10), "CAST('NULL' AS varchar(10))")
+                .addRoundTrip("char(10)", "NULL", createVarcharType(10), "CAST(NULL AS varchar(10))")
+                .addRoundTrip("char(10)", "'简体'", createVarcharType(10), "CAST('简体' AS varchar(10))")
+                .addRoundTrip("char(10)", "'text_a'", createVarcharType(10), "CAST('text_a' AS varchar(10))")
+                .addRoundTrip("char(255)", "'text_b'", createVarcharType(255), "CAST('text_b' AS varchar(255))")
+                .addRoundTrip("char(256)", "'text_c'", createVarcharType(256), "CAST('text_c' AS varchar(256))")
+                .addRoundTrip("char(65535)", "'text_d'", createVarcharType(65535), "CAST('text_d' AS varchar(65535))")
+                .addRoundTrip("char(65536)", "'text_e'", createVarcharType(65536), "CAST('text_e' AS varchar(65536))")
+                .addRoundTrip("char(5)", "'攻殻機動隊'", createVarcharType(5), "CAST('攻殻機動隊' AS varchar(5))")
+                .addRoundTrip("char(32)", "'攻殻機動隊'", createVarcharType(32), "CAST('攻殻機動隊' AS varchar(32))")
+                .addRoundTrip("char(10)", "'😂'", createVarcharType(10), "CAST('😂' AS varchar(10))")
+                .addRoundTrip("char(77)", "'Ну, погоди!'", createVarcharType(77), "CAST('Ну, погоди!' AS varchar(77))")
+                .addRoundTrip("char(77)", "'+-*/!234=1'", createVarcharType(77), "CAST('+-*/!234=1' AS varchar(77))")
+                .execute(getQueryRunner(), igniteCreateAndInsert("test_char"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_char"));
+    }
+
+    @Test
+    public void testVarchar()
```

Comment:
We can merge `testChar` and `testVarchar` into one with a helper method which is similar to `binaryTest`.